lua/plenary/path.lua

```diff
@@ -474,20 +474,77 @@ function Path:rename(opts)
   return status
 end
 
+--- Copy files or folders.
+---@param opts table: options to pass to toggling registered actions
+---@field destination string|Path: target file path to copy to
+---@field recursive bool: whether to copy folders recursively (default: true)
+---@field override bool: whether to override files (default: true)
+---@field interactive bool: confirm if copy would override; precedes `override` (default: false)
+---@field respect_gitignore bool: skip folders ignored by all detected `gitignore`s (default: true)
+---@field hidden bool: whether to add hidden files in recursively copying folders (default: true)
+---@return table {[Path of destination]: bool} indicating success of copy; nested tables constitute sub dirs
 function Path:copy(opts)
   opts = opts or {}
+  opts.recursive = F.if_nil(opts.recursive, false, opts.recursive)
+  opts.override = F.if_nil(opts.override, true, opts.override)
 
+  local dest = opts.destination
   -- handles `.`, `..`, `./`, and `../`
-  if opts.destination:match "^%.%.?/?\\?.+" then
-    opts.destination = {
-      uv.fs_realpath(opts.destination:sub(1, 3)),
-      opts.destination:sub(4, #opts.destination),
+  if not Path.is_path(dest) then
+    if type(dest) == "string" and dest:match "^%.%.?/?\\?.+" then
+      dest = {
+        uv.fs_realpath(dest:sub(1, 3)),
+        dest:sub(4, #dest),
+      }
+    end
+    dest = Path:new(dest)
+  end
+  -- success is true in case file is copied, false otherwise
+  local success = {}
+  if not self:is_dir() then
+    local flags = {}
+    if opts.interactive and dest:exists() then
+      vim.ui.select(
+        { "Yes", "No" },
+        { prompt = string.format("Overwrite existing %s?", dest:absolute()) },
+        function(_, idx)
+          success[dest] = uv.fs_copyfile(self:absolute(), dest:absolute(), { excl = not (idx == 1) }) or false
+        end
+      )
+    else
+      flags.excl = not opts.override
+      -- nil: not overriden if `override = false`
+      success[dest] = uv.fs_copyfile(self:absolute(), dest:absolute(), flags) or false
```

Comment:
do we even need the flags table or can we just do it in the same `fs_copyfile` line ?